pkg/bootstrap/stage.go

```diff
@@ -166,7 +166,19 @@ func extractFromDir(dir, prefix string, img v1.Image, imgName string) error {
 	if err := extract(imgName, tempDir, prefix, r); err != nil {
 		return err
 	}
-	return os.Rename(tempDir, dir)
+	if err := os.Rename(tempDir, dir); err != nil {
+		files, err := ioutil.ReadDir(tempDir)
+		if err != nil {
+			return err
+		}
+		for _, file := range files {
+			if err := os.Rename(filepath.Join(tempDir, file.Name()), filepath.Join(dir, file.Name())); err != nil {
+				os.Remove(filepath.Join(dir, file.Name()))
```

Comment:
We should at least log the errors returned here if any. Possible escalate and stop execution.

Reply:
Could accumulate errors like we do here: https://github.com/rancher/k3s/blob/master/pkg/deploy/controller.go#L121

Reply:
bail on first unrecoverable error condition, imo

Reply:
I kinda like accumulating and then reporting them together so that the user has a complete list of problems, rather than having to fix one thing and then retry so it can fail on the next one.

Reply:
just realized i never hit "send" on my comment! But it was the same thing: these errors need handled.

I'm with @dweomer: just bail on first one. At least in this scenario, I think that gets the job done simply enough without adding the complexity of gathering and properly displaying multiple errors.